src/analyses/apron/octApron.apron.ml

```diff
@@ -97,15 +97,22 @@ struct
         (* TODO: don't go to top, but just forget r *)
         D.top_env (A.env ctx.local)
 
+
+  let assert_fn ctx e change =
+    if not change then
+      ctx.local
+    else
+      let res = D.assert_inv ctx.local e false in
+      if D.is_bot_env res then raise Deadcode;
+      res
+
   let special ctx r f args =
     if D.is_bot ctx.local then D.bot () else
       begin
         match LibraryFunctions.classify f.vname args with
-        | `Assert expression -> ctx.local
         | `Unknown "printf" -> ctx.local
-        | `Unknown "__goblint_check" -> ctx.local
-        | `Unknown "__goblint_commit" -> ctx.local
-        | `Unknown "__goblint_assert" -> ctx.local
+        (* | `Assert (e, _, change) -> assert_fn ctx e change *)
+        | `Assert (e, _, _) -> assert_fn ctx e true (* TODO: make assert() always change *)
```

Comment:
Why? Should this not be consistent across base and octApron?

Reply:
Indeed it should be and I think it would make sense for `assert()` to change by default (if needed we can have an option to disable that). I don't see the reason not to, but I wanted to wait until the GobCon discussion of this before making that change and finalizing this PR.

Reply:
I changed `assert()` refinement to follow `sem.assert.refine` option which is true by default.